ibis/backends/risingwave/__init__.py

```diff
@@ -257,3 +311,364 @@ def _register_in_memory_table(self, op: ops.InMemoryTable) -> None:
             with self.begin() as cur:
                 cur.execute(create_stmt_sql)
                 extras.execute_batch(cur, sql, data, 128)
+
+    def exec_driver_sql(self, query: str):
+        with self.begin() as cur:
+            cur.execute(query)
+
+    @staticmethod
+    def _extract_schema(
+        obj: pd.DataFrame | pa.Table | ir.Table | None = None,
+        schema: ibis.Schema | None = None,
+    ) -> ibis.Schema:
+        if obj is None and schema is None:
+            raise com.IbisError("The schema or obj parameter is required")
+
+        import pandas as pd
+        import pyarrow as pa
+        import pyarrow_hotfix  # noqa: F401
+
+        if isinstance(obj, (pd.DataFrame, pa.Table)):
+            obj = ibis.memtable(obj)
+
+        if obj is not None and schema is not None:
+            if not obj.schema().equals(ibis.schema(schema)):
+                raise com.IbisTypeError(
+                    "Expression schema is not equal to passed schema. "
+                    "Try passing the expression without the schema"
+                )
+        if schema is None:
+            schema = obj.schema()
+        return schema
+
+    def create_materialized_view(
+        self,
+        name: str,
+        obj: ir.Table,
+        *,
+        database: str | None = None,
+        schema: str | None = None,
+        overwrite: bool = False,
+    ) -> ir.Table:
+        """Creating a materialized view. The created materialized view can be accessed like a normal table.
+
+        Parameters
+        ----------
+        name
+            Materialized view name to Create.
+        obj
+            The select statement to materialize.
+        database
+            Name of the database where the view exists, if not the default.
+        schema
+            Name of the schema where the view exists, if not the default
+        overwrite
+            Whether to overwrite the existing materialized view with the same name
+
+        Returns
+        -------
+        Table
+            Table expression
+        """
+        if database is not None and database != self.current_database:
+            raise com.UnsupportedOperationError(
+                f"Creating materialized views in other databases is not supported by {self.name}"
+            )
+        else:
+            database = None
+
+        if overwrite:
+            temp_name = util.gen_name(f"{self.name}_table")
+        else:
+            temp_name = name
+
+        table = sg.table(
+            temp_name, db=schema, catalog=database, quoted=self.compiler.quoted
+        )
+
+        create_stmt = sge.Create(
+            this=table,
+            kind="MATERIALIZED VIEW",
+            expression=self.compile(obj),
+        )
+        self._register_in_memory_tables(obj)
+
+        this = sg.table(name, catalog=database, quoted=self.compiler.quoted)
+        with self._safe_raw_sql(create_stmt) as cur:
+            if overwrite:
+                cur.execute(
```

Comment:
Similar to other comments, can we not use `drop_materialized_view()` for this?

Reply:
👀 Do you mean we should use `drop view`? Or you think we should use `drop_materialized_view()` here.

Reply:
I meant using the appropriate function, ``drop_view()`` or `drop_materialized_view()`, instead of the `sqlglot` expression:
```
sge.Drop(kind="MATERIALIZED VIEW", this=this, exists=True).sql(
                        self.dialect
                    )
```
IMO, `sqlglot` expressions look complex and replacing them with calls to the appropriate functions would improve readability.

Reply:
got it. updated